AISKU/src/AISku.ts

```diff
@@ -200,60 +215,47 @@ export class AppInsightsSku implements IApplicationInsights {
 
             // Will get recalled if any referenced values are changed
             _addUnloadHook(onConfigChange(cfgHandler, () => {
-                let configCs =  _config.connectionString;
-
-                function _parseCs() {
-                    return createSyncPromise<ConnectionString>((resolve, reject) => {
-                        doAwaitResponse(configCs, (res) => {
-                            let curCs = res && res.value;
-                            let parsedCs = null;
-                            if (!res.rejected && curCs) {
-                                // replace cs with resolved values in case of circular promises
-                                _config.connectionString = curCs;
-                                parsedCs = parseConnectionString(curCs);
-                            }
-                            // if can't resolve cs promise, null will be returned
-                            resolve(parsedCs);
-                        });
-                    });
-
-                }
+                let configCs = _config.connectionString;
                 
                 if (isPromiseLike(configCs)) {
                     let ikeyPromise = createSyncPromise<string>((resolve, reject) => {
-                        _parseCs().then((cs) => {
-                            let ikey = _config.instrumentationKey;
-                            ikey = cs && cs.instrumentationkey || ikey;
-                            resolve(ikey);
-                        }).catch((e) => {
-                            // parseCs will always resolve(unless timeout)
-                            // return null in case any error happens
-                            resolve(null);
+                        doAwaitResponse(_parseCs(_config, configCs), (rsp) => {
```

Comment:
Changed to use `doAwaitResponse` as this helps with minification in several ways.
- no `then` or `catch` names (which can't be minified)
- Only 1 function (when using then/catch each one generates a function)